packages/cozy-konnector-libs/libs/linkBankOperations.js

```diff
@@ -170,9 +173,83 @@ class Linker {
       })
     })
     .then(() => {
-      return result
+      const notLinkedBills = this.getNotLinkedBills(result)
+      const billsGroups = this.groupBillsByOriginalDate(notLinkedBills)
+
+      const combinations = flatten(
+        Object
+          .values(billsGroups)
+          .map(billsGroup => this.generateBillsCombinations(billsGroup))
+      )
+
+      const combinedBills = combinations.map(
+        combination => this.combineBills(...combination)
+      )
+
+      return Promise.all(combinedBills.map(bill => {
+        return findDebitOperation(this.cozyClient, bill, options, allOperations)
+      }))
+        .then(debitOperations => debitOperations.filter(Boolean))
+        .then(debitOperations => {
+          debitOperations.forEach((debitOperation, index) => {
+            combinedBills[index].originalBills.forEach(originalBill => {
+              const res = result[originalBill._id]
+              res.debitOperation = debitOperation
+
+              this.addBillToOperation(originalBill, debitOperation)
+            })
+          })
+
+          return result
+        })
+        .catch(err => console.log(err))
     })
   }
+
+  getNotLinkedBills (bills) {
+    const notLinkedBills = Object.values(bills)
+      .filter(bill => !bill.debitOperation)
+      .map(bill => bill.bill)
+
+    return notLinkedBills
+  }
+
+  groupBillsByOriginalDate (bills) {
+    return groupBy(bills, bill => bill.originalDate)
+  }
+
+  generateBillsCombinations (bills) {
+    const combinations = Array.from({ length: bills.length })
+      .reduce((combinations, item, index) => {
+        if (index === 0) {
+          return combinations
+        }
+
+        const newCombinations = []
+
+        const iter = geco.gen(bills.length, index + 1, bills)
+        for (const combination of iter) {
+          newCombinations.push(combination)
+        }
+
+        return [
+          ...combinations,
+          ...newCombinations
+        ]
+    }, [])
+
```

Comment:
It seems a bit complicated.

Why not simply do 

```
let combinations = []
for (let n = 0; n < combinations.length; n++) {
	const combinationsN = geco.gen(bills.length, n + 1, bills)
    combinations = combinations.concat(combinationsN)
}
return combinations
```

This way you do not have the need for the flatten above.